rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -327,7 +346,8 @@ nested_type = '%s__%s__%s' % (field.type.pkg_name, 'msg', field.type.type)
 @[    elif field.type.type == 'string']@
     PyList_SetItem(py@(field.name), idx@(field.name), PyUnicode_FromString(tmpmessagedata@(field.name)[idx@(field.name)].data));
 @[    elif field.type.type == 'bool']@
-    PyList_SetItem(py@(field.name), idx@(field.name), tmpmessagedata@(field.name)[idx@(field.name)] ? Py_True : Py_False);
+@# using PyBool_FromLong because PyList_SetItem will steal ownership of the passed item
+    PyList_SetItem(py@(field.name), idx@(field.name), PyBool_FromLong(tmpmessagedata@(field.name)[idx@(field.name)] ? 1 : 0));
```

Comment:
This was slightly tricky to find :angry:

Same below.

Reply:
pretty unconvenient that we need to use this hack because `Py_False` and `Py_True` don't give the right refcount here... Looking at the doc that seems the be the only suitable workaround :+1:

Reply:
Well, we could also not call `Py_DECREF` for this case - but that would make our template logic more complex. So I would say it is only necessary since we want to treat all types uniformly.

Reply:
Agreed, and I think we agree that the template logic doesnt need additional complexity ;)